### ISSUE TITLE
Executor task fallback config

### ISSUE THREAD

**Issue by tarnishablec** [nrwl/nx#21452]
- [ ] I'd be willing to implement this feature ([contributing guide](https://github.com/nrwl/nx/blob/master/CONTRIBUTING.md))

## Description

When a executor task target fails, another task target can be executed automaticly as fallback.

## Motivation

The building process may sometimes fail for unknown reasons, so a backup process is needed.
For example, sometimes esbuild process wont build successfully, nx should fallback to rollup.

## Suggested Implementation

Add a "fallback" field to target config object 
```json
// project.json
{
  "targets":{
    "esbuild":{
       "executor": "@nx/esbuild:build",
       "fallback": "rollup"
    },
    "rollup":{
       "executor": "@nx/rollup:build",
    }
  }
}
```

**Comment by AgentEnder** [comment]
I don't think this is something we would want to support happening automatically, as it could lead to a workspace degrading overtime without noticing and result in a bad overall maintenance state. Instead, if a project is flaky or can't be successfully built with a given tool I'd recommend renaming the targets to be based on what they are doing rather than the tool they use and swapping over to the new tool only once you've gotten it to work in a consistent manner
